docker-setup.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -e;
+set -u;
+
+localsetup() {
+    rm -rf package-lock.json node_modules
+    npm install
+
+    cd types
+    rm -rf package-lock.json node_modules
+    npm install
+
+    cd ../backend
+    rm -rf package-lock.json node_modules
+    npm install
+
+    cp -R ../types packages/Upgrade
+    cd packages/Upgrade
+    rm -rf package-lock.json node_modules
+    npm install
+
+    cd ../../../frontend
+    rm -rf package-lock.json node_modules
+    npm install
```

Comment:
All these `npm install`s mean that after I run `make setup-local`, `git status` is not clean.  IMO that's not a great developer experience -- we should run `npm ci` in this script instead.

Choosing to consume new minor/patch/etc. versions of dependencies should be an explicit thing that a developer plans/chooses to do, not something that can happen by accident during local environment setup.

Reply:
ok I can update the script and look into how to obtain the right version with nix